docs/WIRELESS.md

```diff
@@ -188,6 +188,55 @@ as the primary slot and the user has not explicitly set the slot index via the c
 automatically designates the first physical slot (with the lowest index) as the primary option.
 EVE will only attempt to connect with the eSIM if the user explicitly selects the eSIM slot.
 
+### Modem (Un)reliability and recovery mechanisms in EVE OS
+
+Modern cellular modems are critical for connectivity but often suffer from instability
+due to firmware, driver, or management software issues. EVE OS incorporates reactive
+recovery mechanisms to mitigate these issues and stabilize device connectivity.
+
+**Main challenges in modem stability**:
+
+* Unreliable modem firmware: proprietary and vendor-supplied modem firmware often contains
+  bugs that can lead to crashes or unresponsiveness. These issues are outside the control
+  of the OS and require recovery strategies.
+
+* Community-maintained drivers: Linux kernel drivers for modem control and data-plane
+  protocols (QMI, MBIM, MHI) are developed by the open-source community rather than
+  the modem manufacturers. This can result in compatibility gaps, occasional breakage,
+  or incomplete error handling.
+
+* ModemManager limitations/bugs: the ModemManager daemon may contain internal bugs
+  or fail to properly recognize and reinitialize modem after it rebooted due to firmware
+  issues.
+
+**EVE OS recovery strategies**:
```

Comment:
Is the assumption that the user enables at most one of the three strategies? If not, what happens if multiple are enabled? Do they get tried in some sequence? 
If it is a choice between between none or one of the three it would be less error prone to use an integer in the range [0-3] to choose the behavior.

Reply:
User can enable any subset of these options. Below is explained that driver reloading and ModemManager restart are done together (or just one of them if the other is not enabled) and before watchdog is triggered.